src/pyvcam_driver.py

```diff
@@ -0,0 +1,537 @@
+        """
+        Changes exposure resolution.
+        Keys:
+            0 - One Millisecond
+            1 - One Microsecond
+
+        :param key_or_value: Key or value to change.
+        :return: None.
+        """
+        self.cam.exp_res = key_or_value
+
+    def exp_res_index(self) -> int:
+        """
+        Returns current exposure resolution index.
+
+        :return:
+        """
+        return self.cam.exp_res_index
+
+    def exp_resolutions(self) -> dict[str, int]:
+        """
+        Returns dict containing exposure resolutions supported by the camera.
+
+        :return:
+        """
+        return dict(self.cam.exp_resolutions)
+
+    def exp_time(self) -> int:
+        """
+        Returns the exposure time the camera will use if not given an exposure time. 
+        It is recommended to modify this value to modify your acquisitions for better abstraction.
+
+        :return:
+        """
+        return self.cam.exp_time
+    
+    def set_exp_time(self, value: int) -> None:
+        """
+        Changes exposure time.
+
+        :param value: Desired exposure time
+        :return: None.
+        """
+        self.cam.exp_time = value
+
+    def gain(self) -> int:
+        """
+        Returns the current gain index for a camera. 
+        A ValueError will be raised if an invalid gain index is supplied to the setter.
+
+        :return:
+        """
+        return self.cam.gain
+
+    def set_gain(self, value: int) -> None:
+        """
+        Changes gain.
+
+        :param value: Desired gain value
+        :return: None.
+        """
+        self.cam.gain = value
+
+    def last_exp_time(self) -> int:
+        """
+        Returns the last exposure time the camera used for the last successful 
+        non-variable timed mode acquisition in what ever time resolution it was captured at.
+
+        :return:
+        """
+        return self.cam.last_exp_time
+
+    def pix_time(self) -> int:
+        """
+        Returns the camera's pixel time, which is the inverse of the speed of the camera. 
+        Pixel time cannot be changed directly; instead users must select a desired speed table index value that has the desired pixel time. 
+        Note that a camera may have additional speed table entries for different readout ports. 
+        See Port and Speed Choices section inside the PVCAM User Manual for a visual representation of a speed table and to see which 
+        settings are controlled by which speed table index is currently selected.
+
+        :return:
+        """
+        return self.cam.pix_time
+
+    def port_speed_gain_table(self) -> dict:
+        """
+        Returns a dictionary containing the port, speed and gain table, 
+        which gives information such as bit depth and pixel time for each readout port, speed index and gain.
+
+        :return:
+        """
+        return self.cam.port_speed_gain_table
+
+    def readout_port(self) -> int:
```

Comment:
```suggestion
    def get_readout_port(self) -> int:
```
For symmetry with `set_readout_port`.